image/autodetect.go

```diff
@@ -103,8 +102,8 @@ func Autodetect(path string) (string, error) {
 	case header.MediaType == string(schema.MediaTypeManifestList):
 		return TypeManifestList, nil
 
-	case header.MediaType == "" && header.SchemaVersion == 0 && header.Config != nil:
-		// config files don't have mediaType/schemaVersion header
+	case header.RootFS != nil:
```

Comment:
I think it is good to check `rootfs`(required) instead `config`(optional).
However, statement `header.MediaType == "" && header.SchemaVersion == 0` had better to be kept, which are also terms differ with other json descriptors.